### PR TITLE
GCW 1619 prevent not inventoried item from being designated 

### PR DESCRIPTION
**Pull request by trojanh**

### CODE REVIEW

File: app/models/orders_package.rb [hunk 1]
```diff
@@ -162,4 +164,14 @@ def update_designation_of_package
   def destroy_stockit_record(operation)
     StockitSyncOrdersPackageJob.perform_now(package.id, self.id, operation) unless package.is_singleton_package?
   end
+
+  def valid_package_state?
+    package.received? && package.inventory_number? && package.locations.present?
```
Comment (steveyken) [inline app/models/orders_package.rb]: could we split these up into separate validations?

    validates :inventory_number, :presense
    validates :locations, :presense

Comment (trojanh) [inline app/models/orders_package.rb]: Yeah that will make the code more readable. Thanks @steveyken

File: app/models/orders_package.rb [hunk 2]
```diff
@@ -162,4 +166,22 @@ def update_designation_of_package
   def destroy_stockit_record(operation)
     StockitSyncOrdersPackageJob.perform_now(package.id, self.id, operation) unless package.is_singleton_package?
   end
+
+  def package_inventory_number
+    if !package.inventory_number?
```
Comment (namrataukirde) [inline app/models/orders_package.rb]: Can we use unless here instead of if ! ?

File: app/models/orders_package.rb [hunk 3]
```diff
@@ -3,7 +3,11 @@ class OrdersPackage < ActiveRecord::Base
   belongs_to :package
   belongs_to :updated_by, class_name: 'User'
 
+  validate :package_inventory_number, :on => :create
```
Comment (namrataukirde) [inline app/models/orders_package.rb]: Use new syntax here.  on:  :create

Comment (trojanh) [inline app/models/orders_package.rb]: :+1:

File: app/models/orders_package.rb (hunk 2 above)
Comment (trojanh) [inline app/models/orders_package.rb]: :+1:

**Comment by namrataukirde** [comment]
Closing this PR as we are not currently working on it.

**Pull request closed without merging**